lib/foreman/dynflow.rb

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+require 'dynflow'
+
+module Foreman
+  # Class for configuring and preparing the Dynflow runtime environment.
+  class Dynflow
+    require File.expand_path('../dynflow/configuration', __FILE__)
+    require File.expand_path('../dynflow/persistence', __FILE__)
+    require File.expand_path('../dynflow/daemon', __FILE__)
+
+    def initialize
+      @required = false
+    end
+
+    def config
+      @config ||= Foreman::Dynflow::Configuration.new
+    end
+
+    # call this method if your engine uses Dynflow
+    def require!
+      @required = true
+    end
+
+    def required?
+      @required
+    end
+
+    def initialized?
+      !@world.nil?
+    end
+
+    def initialize!
+      return unless @required
+      return @world if @world
+
+      if config.lazy_initialization && defined?(PhusionPassenger)
```

Comment:
I do wonder if we should keep supporting this use case, now that we have foreman, we can always install dynflow service when foreman starts.

I don't know if we are actually gaining anything by running it in the same process vs using it more similar to how its used in production?